deploy/crds/hco00.crd.yaml

```diff
@@ -179,6 +179,193 @@ spec:
                 description: BareMetalPlatform indicates whether the infrastructure
                   is baremetal.
                 type: boolean
+              hyperConvergedConfig:
+                description: Let the user define parameters that will be propagated
+                  to the CRs of managed components
+                properties:
+                  infraNodeSelector:
```

Comment:
How about placing this in a `placement` met-struct?

Something like:

```
placement:
    controlPlane:
        nodeSelector:
        tolerations:
    workloads:
       nodeSelector:
       tolerations:
```

In general I miss tolerations here. Further you will have to find out, how things will behave if we just set tolerations. Will the default tolerations still be applied in addition? If not, will that influence our API design?

Reply:
Yes, I agree that a nested struct looks better.
Why do you think that we have to skip tolerations is we have just them?

Reply:
> Why do you think that we have to skip tolerations is we have just them?

I think we should be well aware of what will happen if someone adds a toleration or a node-selector. Especially on tolerations there are normally a bunch of default tolerations added. We should just explicitly define what we expect to happen. Things have to behave exactly the same for all our components in kubevirt, or we end up with e.g. DaemonSets being scheduled inconsistently caross different kubevirt parts.

Reply:
This latest change transitions the term `infra` to `controlPlane`.  are we concerned at all about the fact that portions of our controlPlane (like virt-handler) need to run where workloads run? That was part of the reasoning for `infra` as a term

Reply:
I don't understand why `infra` is less ambiguous. `virt-handler` would not be part of `infra` and it would not be part of `controlPlane` here. Just to reiterate the requirment: We want placement control for our cluster-wide components  and independent of that a general filter for workload placement. Workload placements in most cases means influencing the daemonsets. One thing which came up repeatedly is that people do not only want to not run VMs on specific nodes, they expect at the same moment that our node-related infra is not there as well.

This is mostly for CDI a little bit more tricky, since in general nothing speaks about  running an unprivileged pod on all workers for imports. Here I am not exactly sure what to do.

But in general `controlPlane` or `infra` as used here includes:

 * controllers
 * apiservers and
 * webhooks

with other words pretty much solely `deployments`.

`workloads` mostly means our infra which is required on nodes which normally are `daemonsets`.


Alternative terms could be `clusterInfra` and `nodeInfra`, where `nodeInfra` with the placement of our components of course always adds a pre-filter on wher VMs can run.

Reply:
Operators placement will be determined on the OLM subscription object (once implemented on the OLM side) so maybe controlPlane here is too much.

Reply:
> I don't understand why infra is less ambiguous. virt-handler would not be part of infra and it would not be part of controlPlane here. 

yeah, I'm not convinced the naming really matters too much now. All the options thrown out so far would work. @tiraboschi I'll support whatever terms you come up with.

Reply:
> yeah, I'm not convinced the naming really matters too much now. All the options thrown out so far would work. @tiraboschi I'll support whatever terms you come up with.

Same here. I mostly care that we have a description on top then which fits the above scenarios.

Reply:
I reverted back to infra and I renamed from tolerations to extraTolerations to make more evident that they are going to be only additions to the set of default tolerations.